src/main/java/org/jabref/gui/edit/CopyMoreAction.java

```diff
@@ -145,6 +146,33 @@ private void copyDoi() {
         }
     }
 
+    private void copyDoiUrl() {
+        List<BibEntry> entries = stateManager.getSelectedEntries();
+
+        // Collect all non-null DOI urls.
+        List<String> dois = entries.stream()
+                .filter(entry -> entry.getDOI().isPresent())
+                .map(entry -> entry.getDOI().get().getURIAsASCIIString())
+                .collect(Collectors.toList());
+
+        if (dois.isEmpty()) {
```

Comment:
In general looks good so far but now you have a lot of code duplication with the other method.
So why not create a common method where you just have the list of dois/doi urls as input parameter?

Reply:
I overlooked this, thank you for the feedback. I will change this.